file_util/src/file.cc

```diff
@@ -27,23 +27,32 @@ namespace boost_file = boost::filesystem;
         return true;
     }
 
-    string read(string const& path) {
-        string contents;
-        if (!read(path, contents)) {
-            return {};
-        }
-        return contents;
-    }
+    // This function attempts to read a file at the given path in text mode and return its contents as
+    // a string. If the file at the path is a symlink, it will be followed until a non-symlink file is
+    // found. If anything goes wrong, a file_error will be thrown.
+    std::string read(const boost_file::path& path) {
+      boost_file::path p(path);
+      while (boost_file::is_symlink(p)) {
```

Comment:
Is the dereferencing of the symlink necessary?  Opening the symlink should follow, right?